tests/tests/test_fault_tolerance.py

```diff
@@ -224,7 +224,19 @@ def test_image_download_retry_timeout(
             if test_set["blockAfterStart"]:
                 # Block after we start the download.
                 deployment_id, new_yocto_id = common_update_procedure(valid_image)
-                mender_device.run("fuser -mv %s" % (inactive_part))
+                starttime = time.time()
+                while starttime + 60 * 10 >= time.time():
+                    output = mender_device.run(
+                        "fuser -mv %s" % (inactive_part), warn_only=True
+                    )
+                    if output.rstrip() != "":
+                        # If it succeeds, stop.
+                        break
+                    else:
+                        time.sleep(0.1)
```

Comment:
Hmm, I'm a little bit worried about this change. I know that the reason for the exponential backoff is that for emulated QEMU CPUs, if the system is bogged down, it can get so slow that short timeouts like this essentially lead to no other work being performed, because SSH takes up all the processing. I recently saw this while developing the update control tests, so this is definitely still an issue. The backoff handles these situations by slowing down and leaving larger and larger gaps for other work to happen.

I have a suggestion for a fix which is more "backoff tolerant": Add an `ArtifactInstall_Enter_00` state script to the artifact, which creates a file, and then sleeps until that file is removed. You can get look for that file with `run()`, when it is present you manipulate the network settings, and then remove it for the client to continue.

How does that sound?

Reply:
That sounds like a much better approach, thanks! I'll rework this a a later time